File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
// Lv 2. 올바른 괄호 - 스택/큐
function solution(s){
    const stack = [s[0]];
    
    for(let i = 1; i < s.length; i++){
        if(stack[stack.length-1] === '(' && s[i] === ')'){
            stack.pop();
        }else{
            stack.push(s[i]);
        }
    }
    
    return stack.length === 0 ? true : false;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    arrow = 0;
    
    for i in s:
        if(i == '('):
            arrow += 1
        elif(i == ')'):
            arrow -= 1
            
        if(arrow < 0):
            return False
    
    if(arrow == 0):
        return True
    else:
        return False
